src/java/org/apache/cassandra/index/sai/disk/vector/VectorMemtableIndex.java

```diff
@@ -372,13 +370,53 @@ protected PrimaryKey computeNext()
         }
     }
 
+    private class ListRangeIterator extends RangeIterator
```

Comment:
WFM either way, but I'm modestly in favor of move this to a top level class w/ a docstring -- seems like it could be more generally useful.

Reply:
Already found another use for it in the `limitToTopResults` optimization :)